jdaviz/core/template_mixin.py

```diff
@@ -212,6 +213,35 @@ def _remove_viewer_callbacks(self, viewer_id):
                                   if k.split(':')[0] != viewer_id}
 
 
+def skip_if_no_updates_since_last_active(skip_if_not_active=True):
+    def decorator(meth):
+        def wrapper(self, msg={}):
+            if isinstance(msg, dict) and msg.get('name', None) == 'is_active':
+                if self.is_active and meth.__name__ in self._methods_skip_since_last_active:
+                    # then we haven't received any other messages since the last time the plugin
+                    # received an is_active switch, and so we should skip calling the method.
+                    return
+            elif not self.is_active:
+                # then we've received some other message while the plugin is inactive.
+                # Next time the plugin becomes active we want to call the wrapped method,
+                # so we'll remove from the skip list.
+                if meth.__name__ in self._methods_skip_since_last_active:
+                    self._methods_skip_since_last_active.remove(meth.__name__)
+
+            if skip_if_not_active and not self.is_active:
+                return
+
+            # call the method as normal, and add it to the skip list (to be skipped if is_active
+            # toggles before any *other* messages are received)
+            ret_ = meth(self, msg)
+            if meth.__name__ not in self._methods_skip_since_last_active:
+                self._methods_skip_since_last_active.append(meth.__name__)
```

Comment:
Should this append happen before `meth(...)` is invoked, in case it is an intensive method that takes a long time?

Reply:
that's a good idea.... my only concern is what happens if there is an exception in `meth(...)`.  But maybe we could wrap `meth(...)` in a try/except to remove from the list if it fails (and then raise the original error)?  I'll think if there would be any unintended consequences of that, but I can see how otherwise this could still technically result in a (small, but unnecessary) queue of calls.